src/mongocxx/test/uri.cpp

```diff
@@ -41,4 +43,36 @@ TEST_CASE("URI", "[uri]") {
             REQUIRE(e.code() == mongocxx::error_code::k_invalid_uri);
         }
     }
+
+    SECTION("Empty fields") {
+        mongocxx::uri u{};
+
+        // Values that should be empty with a blank URI
+        REQUIRE(u.auth_mechanism() == "");
+        REQUIRE(u.database() == "");
+        REQUIRE(bsoncxx::to_json(u.options()) == "{ }");
+        REQUIRE(u.password() == "");
+        REQUIRE(u.read_concern().acknowledge_string().empty());
+        REQUIRE(!u.read_preference().tags());
+        REQUIRE(!u.read_preference().max_staleness());
+        REQUIRE(u.replica_set() == "");
+        REQUIRE(u.username() == "");
+        REQUIRE(!u.write_concern().nodes());
+        
+        // Values with defaults with a blank URI
+        REQUIRE(u.ssl() == false);
```

Comment:
I don't know that these values are necessarily correct. I have simply verified that (1) they aren't brazenly unreasonable to my untrained eye, and (2) after the fix below, none of these accessor functions segfault :-)